builder/src/communication.js

```diff
@@ -0,0 +1,93 @@
+const rp = require('request-promise')
+const Buffer = require('./buffer')
+
+class Communication {
+  constructor (config) {
+    this.config = config
+    this.resource = this.config.build_api_url + 'build/'
+    this.startEndpoint = '/start'
+    this.stdoutEndpoint = '/stdout'
+    this.stderrEndpoint = '/stderr'
+    this.endEndpoint = '/end'
+    this.stdoutBuffer = new Buffer()
+    this.stderrBuffer = new Buffer()
+  }
+
+  async _req (uri, body) {
+    const options = {
+      method: 'PUT',
+      uri,
+      json: (body && true) || false,
```

Comment:
Why `&& true`?

Reply:
the json property is used to say the body is in json format. However if there is no body (null) and the property json is set to true, the client (api) will try to parse the json body (which is null) and will throw an error. So before saying if the body is in json we check that there is a body

Reply:
Why not using `body !== null`?

Reply:
Fixed.